### PR TITLE
MCLOUD-6712: Fix error method parameters for DatabaseSplitConnection validator

### PR DESCRIPTION
**Pull request by NadiyaS**
### Description

Error code `Error::WARN_WRONG_SPLIT_DB_CONFIG` was added to `sprintf()` method instead of `error()` in DatabaseSplitConnection validator 

### Fixed Issues (if relevant)

1. https://jira.corp.magento.com/browse/MCLOUD-6712

### Manual testing scenarios

- add custom DB split connection to DATABASE_CONFIGURATION.
- deploy should be passed but warning exists in cloud.error.log file with code 2002 and corresponding message

### Release notes

For user-facing changes, add a meaningful release note. For examples, see [Magento Cloud ECE-tools release notes](https://devdocs.magento.com/cloud/release-notes/ece-release-notes.html).

### Associated documentation updates

Add link to Magento DevDocs PR or Issue, if needed.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] Pull request introduces user-facing changes and includes meaningful updates for any required release notes and documentation changes
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by meker12** [comment]
running tests

File: src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php [hunk 1]
```diff
@@ -78,14 +79,19 @@ public function testMessageValidate()
...
+                'Split database configuration was detected in the property DATABASE_CONFIGURATION of the'
+                    . ' file .magento.env.yaml:' . PHP_EOL
+                    . '- connection: checkout' . PHP_EOL
+                    . '- connection: sales' . PHP_EOL
+                    . '- slave_connection: checkout' . PHP_EOL
+                    . '- slave_connection: sales' . PHP_EOL
+                    . 'Magento Cloud does not support a custom split database configuration,'
+                    . ' such configurations will be ignored',
+                'Remove custom connections for split databases from DATABASE_CONFIGURATION variable '
+                  . 'in .magento.env.yam',
```
Comment (BarnyShergold) [inline src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php]: incorrect file name - missed of the 'l' of yaml

File: src/Config/Validator/Deploy/DatabaseSplitConnection.php [hunk 2]
```diff
@@ -63,15 +63,21 @@ public function validate(): Validator\ResultInterface
...
+                'Split database configuration was detected in the property %s'
+                . ' of the file .magento.env.yaml:' . PHP_EOL
+                . '%s' . PHP_EOL
+                . 'Magento Cloud does not support a custom split database configuration,'
+                . ' such configurations will be ignored',
+                DeployInterface::VAR_DATABASE_CONFIGURATION,
+                implode(PHP_EOL, $messageItem)
+            ),
+            sprintf(
+                'Remove custom connections for split databases from %s variable in .magento.env.yam',
```
Comment (BarnyShergold) [inline src/Config/Validator/Deploy/DatabaseSplitConnection.php]: incorrect file name - missed of the 'l' of yaml

Comment (meker12) [inline src/Config/Validator/Deploy/DatabaseSplitConnection.php]:
```suggestion
                'Remove custom connections for split databases from the %s variable in .magento.env.yaml',
```

File: src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php [hunk 3]
```diff
@@ -78,14 +79,19 @@ public function testMessageValidate()
...
-            ->with('Split database configuration was detected in the property DATABASE_CONFIGURATION of the'
-                . ' file .magento.env.yaml:' . PHP_EOL
-                . '- connection: checkout' . PHP_EOL
-                . '- connection: sales' . PHP_EOL
-                . '- slave_connection: checkout' . PHP_EOL
-                . '- slave_connection: sales' . PHP_EOL
-                . 'Magento Cloud does not support a custom split database configuration,'
-                . ' such configurations will be ignored');
+            ->with(
+                'Split database configuration was detected in the property DATABASE_CONFIGURATION of the'
```
Comment (meker12) [inline src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php]:
```suggestion
                'Detected split database configuration in the DATABASE_CONFIGURATION property of the'
```

File: src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php [hunk 4]
```diff
@@ -78,14 +79,19 @@ public function testMessageValidate()
...
+                    . ' file .magento.env.yaml:' . PHP_EOL
+                    . '- connection: checkout' . PHP_EOL
+                    . '- connection: sales' . PHP_EOL
+                    . '- slave_connection: checkout' . PHP_EOL
+                    . '- slave_connection: sales' . PHP_EOL
+                    . 'Magento Cloud does not support a custom split database configuration,'
```
Comment (meker12) [inline src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php]: Not totally clear on this error message-- Is it:  Magento Cloud does not support custom connections in the split database configuration?

Or you can't include split db configuration in the DATABASE_CONFIGURATION?

File: src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php (hunk 1 above)
Comment (meker12) [inline src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php]: So not sure of context --  Remove custom connections for split databases from the ...  ==> That is the suggestion for the user?

Comment (meker12) [inline src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php]:
```suggestion
                  . 'in .magento.env.yaml',
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Need to fix test errors

File: src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php (hunk 1 above)
Comment (NadiyaS) [reply src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php]: Hi Barny, seems like you added comment to outdated code as those places were already fixed

Comment (BarnyShergold) [reply src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php]: I can see it corrected now - all good

**Review by oshmyheliuk — DISMISSED** [review]

File: src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php (hunk 4 above)
Comment (NadiyaS) [reply src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php]: Customer cannot set custom split connections

Comment (arhiopterecs) [reply src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php]: This means that we do not support split database settings with third-party database servers

File: src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php [hunk 5]
```diff
@@ -78,14 +79,19 @@ public function testMessageValidate()
...
-                . 'Magento Cloud does not support a custom split database configuration,'
-                . ' such configurations will be ignored');
+            ->with(
+                'Detected split database configuration in the DATABASE_CONFIGURATION property of the'
+                    . ' file .magento.env.yaml:' . PHP_EOL
+                    . '- connection: checkout' . PHP_EOL
+                    . '- connection: sales' . PHP_EOL
+                    . '- slave_connection: checkout' . PHP_EOL
+                    . '- slave_connection: sales' . PHP_EOL
+                    . 'Magento Cloud does not support a custom split database configuration,'
```
Comment (meker12) [inline src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php]: Consider the following suggestion.

```suggestion
                    . 'Magento Cloud does not support custom connections in the split database configuration,'
```

File: src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php [hunk 6]
```diff
@@ -78,14 +79,19 @@ public function testMessageValidate()
...
+                    . ' such configurations will be ignored',
+                'Remove custom connections for split databases from the DATABASE_CONFIGURATION variable '
```
Comment (meker12) [inline src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php]:
```suggestion
                'Update the DATABASE_CONFIGURATION variable in the `.magento.env.yaml` file to remove custom connections for split databases.'
```

Aren't the only valid connections "checkout" and "sales".   Should the error message be more specific.  Magento Cloud supports split database connections for `checkout` and `sales`  databases only?

File: src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php [hunk 7]
```diff
@@ -78,14 +79,19 @@ public function testMessageValidate()
...
+                  . 'in .magento.env.yaml',
```
Comment (meker12) [inline src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php]: If you accept above suggestion, line 92 not needed.

File: src/Config/Validator/Deploy/DatabaseSplitConnection.php [hunk 8]
```diff
@@ -63,15 +63,20 @@ public function validate(): Validator\ResultInterface
...
-            . 'Magento Cloud does not support a custom split database configuration,'
-            . ' such configurations will be ignored',
-            DeployInterface::VAR_DATABASE_CONFIGURATION,
-            implode(PHP_EOL, $messageItem),
+        return $this->resultFactory->error(
+            sprintf(
+                'Detected split database configuration in the %s property of the file .magento.env.yaml:' . PHP_EOL
+                . '%s' . PHP_EOL
+                . 'Magento Cloud does not support a custom split database configuration,'
+                . ' such configurations will be ignored',
```
Comment (meker12) [inline src/Config/Validator/Deploy/DatabaseSplitConnection.php]:
```suggestion
                . 'Magento Cloud does not support custom connections in the split database configuration,'
                . ' Custom connections will be ignored',
```

File: src/Config/Validator/Deploy/DatabaseSplitConnection.php [hunk 9]
```diff
@@ -63,15 +63,20 @@ public function validate(): Validator\ResultInterface
...
+                DeployInterface::VAR_DATABASE_CONFIGURATION,
+                implode(PHP_EOL, $messageItem)
+            ),
+            sprintf(
+                'Remove custom connections for split databases from the %s variable in .magento.env.yaml',
```
Comment (meker12) [inline src/Config/Validator/Deploy/DatabaseSplitConnection.php]:
```suggestion
                'Update the %s variable in the .magento.env.yaml file to remove custom connections from the split database configuration.',
```

Aren't the only valid connections "checkout" and "sales". Should the error message be more specific? For example:  Magento Cloud supports split database connections for `checkout` and `sales` databases only?

File: src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php [hunk 10]
```diff
@@ -78,14 +79,19 @@ public function testMessageValidate()
...
-                . ' such configurations will be ignored');
+            ->with(
+                'Detected split database configuration in the DATABASE_CONFIGURATION property of the'
+                    . ' file .magento.env.yaml:' . PHP_EOL
+                    . '- connection: checkout' . PHP_EOL
+                    . '- connection: sales' . PHP_EOL
+                    . '- slave_connection: checkout' . PHP_EOL
+                    . '- slave_connection: sales' . PHP_EOL
+                    . 'Magento Cloud does not support a custom split database configuration,'
+                    . ' such configurations will be ignored',
```
Comment (meker12) [inline src/Test/Unit/Config/Validator/Deploy/DatabaseSplitConnectionTest.php]:
```suggestion
                    . ' Custom connections will be ignored',
```

**Review by bbatsche — DISMISSED** [review]

**Review by bbatsche — APPROVED** [review]

**Review by oshmyheliuk — APPROVED** [review]

File: src/Config/Validator/Deploy/DatabaseSplitConnection.php (hunk 2 above)
Comment (andriyShevtsov) [reply src/Config/Validator/Deploy/DatabaseSplitConnection.php]: @BarnyShergold Looks like it was fixed. Please approve

**Review by BarnyShergold — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
QA approved

**Pull request merged**